cmd/dep/main.go

```diff
@@ -84,40 +85,72 @@ func (c *Config) Run() (exitCode int) {
 		},
 	}
 
-	outLogger := log.New(c.Stdout, "", 0)
-	errLogger := log.New(c.Stderr, "", 0)
-
-	usage := func() {
-		errLogger.Println("dep is a tool for managing dependencies for Go projects")
-		errLogger.Println()
-		errLogger.Println("Usage: dep <command>")
-		errLogger.Println()
-		errLogger.Println("Commands:")
-		errLogger.Println()
-		w := tabwriter.NewWriter(c.Stderr, 0, 4, 2, ' ', 0)
+	usage := func(w io.Writer) {
+		fmt.Fprintln(w, "Dep is a tool for managing dependencies for Go projects")
+		fmt.Fprintln(w)
+		fmt.Fprintln(w, "Usage: \"dep [command]\"")
+		fmt.Fprintln(w)
+		fmt.Fprintln(w, "Commands:")
+		fmt.Fprintln(w)
+		tw := tabwriter.NewWriter(w, 0, 4, 2, ' ', 0)
 		for _, cmd := range commands {
 			if !cmd.Hidden() {
-				fmt.Fprintf(w, "\t%s\t%s\n", cmd.Name(), cmd.ShortHelp())
+				fmt.Fprintf(tw, "\t%s\t%s\n", cmd.Name(), cmd.ShortHelp())
 			}
 		}
-		w.Flush()
-		errLogger.Println()
-		errLogger.Println("Examples:")
+		tw.Flush()
+		fmt.Fprintln(w)
+		fmt.Fprintln(w, "Examples:")
 		for _, example := range examples {
-			fmt.Fprintf(w, "\t%s\t%s\n", example[0], example[1])
+			fmt.Fprintf(tw, "\t%s\t%s\n", example[0], example[1])
 		}
-		w.Flush()
-		errLogger.Println()
-		errLogger.Println("Use \"dep help [command]\" for more information about a command.")
+		tw.Flush()
+		fmt.Fprintln(w)
+		fmt.Fprintln(w, "Use \"dep help [command]\" for more information about a command.")
 	}
 
 	cmdName, printCommandHelp, exit := parseArgs(c.Args)
 	if exit {
-		usage()
+		usage(c.Stderr)
 		exitCode = 1
 		return
 	}
 
+	// 'dep help documentation' generates doc.go.
```

Comment:
is this the way that cmd/go does it? seems kinda awkward to wire it in this way.

Reply:
Yep https://golang.org/src/cmd/go/internal/help/help.go#L36